packages/core/src/middleware/koa-quota-guard.ts

```diff
@@ -0,0 +1,76 @@
+import type { MiddlewareType } from 'koa';
+
+import { EnvSet } from '#src/env-set/index.js';
+import RequestError from '#src/errors/RequestError/index.js';
+import { type CloudConnectionLibrary } from '#src/libraries/cloud-connection.js';
+import type Queries from '#src/tenants/Queries.js';
+import assertThat from '#src/utils/assert-that.js';
+import { getTenantSubscriptionPlan } from '#src/utils/subscription/index.js';
+import { type SubscriptionQuota } from '#src/utils/subscription/types.js';
+
+type UsageGuardConfig = {
+  key: keyof SubscriptionQuota;
+  cloudConnection: CloudConnectionLibrary;
+  queries: Queries;
+};
+
+const getTenantUsage = async (key: keyof SubscriptionQuota, queries: Queries): Promise<number> => {
+  if (key === 'applicationsLimit') {
+    const { count } = await queries.applications.findTotalNumberOfApplications();
```

Comment:
Now the m2m apps are included in the applications, but they're respectively limited.

Reply:
updated